Initial commit HW25

jetyyya opened the pull request:
Added: Homework for lesson 25

jetyyya committed:
Initial commit HW25
Added: Homework for lesson 25

jetyyya committed:
Initial commit HW26
Added: Homework for lesson 26
(Хотів ще додати, щоб комп'ютер не мог використовувати назви міст, які вводив користувач, але й так багато часу забрало)

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,63 @@
+                String city = cities[i];
+                if (Character.toUpperCase(city.charAt(0)) == lastLetter) {
+                    System.out.println("Місто: " + city);
+                    lastLetter = Character.toUpperCase(city.charAt(city.length() - 1));
```
Вище вже ініціалізовано дану змінну ідентичними параметрами.
Навіщо дублювати виклик функції? Чи не краще перевикористати змінну, яка була раніше проініціалізована?

jetyyya replied:
Мені здавалось, що для циклу треба окрему змінну, для наочності - виправлю

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,63 @@
+                    System.out.println("Місто: " + city);
+                    lastLetter = Character.toUpperCase(city.charAt(city.length() - 1));
+                    cities[i] = "used";
+                    foundCity = true;
```
Замість використання прапору всередині алгоритму, краще цей метод  розбити на 2 окремі методи

jetyyya replied:
Ми тоді ще не розмовляли про модульність, але так прапори зло :)

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,63 @@
+                }
+            }
+
+            if (!foundCity) {
```
Див. коментар вище

jetyyya replied:
+

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,63 @@
+
+            System.out.println("Ваш хід: (або введіть 'exit', щоб завершити гру)");
+            String playerCity = scanner.nextLine();
+            if (playerCity.equalsIgnoreCase("exit")) {
```
Краще дану умову перенести до умови циклу while, що розташований вище

jetyyya replied:
не зрозумів, усі умови у мене у циклі while... вони брейкають цикл, як вони можуть бути не в ньому?

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,63 @@
+                break;
+            }
+            char firstLetter = Character.toUpperCase(playerCity.charAt(0));
+            if (firstLetter != lastLetter) {
```
Краще всі умови завершення циклу прописати у циклі while, що розміщений вище

jetyyya replied:
не зрозумів, усі умови у мене у циклі while... вони брейкають цикл, як вони можуть бути не в ньому?

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,63 @@
+            lastLetter = Character.toUpperCase(playerCity.charAt(playerCity.length() - 1));
+        }
+        System.out.println("Гра закінчена");
+    }
```
Краще, виділити із даного циклу окремі методи

jetyyya replied:
щаз перероблю...

jetyyya committed:
Second commit HW25
Added: Correction Homework for lesson 25. Moved logic and output in different methods.

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,101 @@
+        printGameInstructions();
+
+        String lastCity = scanner.nextLine();
+        char lastLetter = Character.toUpperCase(lastCity.charAt(lastCity.length() - 1));
```
Краще тут скористатися стандартними функціями String

Bibi4kovIgor commented on HW26/src/main/java/com/alevel/operators/HW26GameOfTheCity.java:
```diff
@@ -0,0 +1,101 @@
+        String lastCity = scanner.nextLine();
+        char lastLetter = Character.toUpperCase(lastCity.charAt(lastCity.length() - 1));
+
+        while (true) {
```
gameplay краще винести в окремий метод

Bibi4kovIgor reviewed (commented):
Виправ, пліз всі зауваження та відміть як "resolved"
